### PR TITLE
Add support for @unpack, @pack! for file-interface (+Test,Documentation)

### PR DESCRIPTION
**Pull request by lukasgrunwald**
Add optional support for @unpack, @pack! from [UnPack.jl](https://github.com/mauro3/UnPack.jl) for file-interface (see #454). They work for the normal file-interface and for `JLD2.Group`'s. The feature can be used as

```julia
using JLD2, UnPack
file = jldopen("example.jld2", "w")
x, y = rand(2)

@pack! file = x, y # equivalent to file["x"] = x; file["y"] = y
@unpack x, y = file # equivalent to x = file["x"]; y = file["y"]
```

The new methods are only added if the user loads the UnPack package himself. This is implemented via [Requires.jl](https://github.com/JuliaPackaging/Requires.jl). For Julia $\geq 1.9$ there are [Package extensions](https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)), but since 1.9 is still in beta, I chose the Requires.jl version. The setup (with the extension living in `ext/UnPackExt.jl`) is compatible with the new "Extension-standard" and requires minimal updates, once this feature is stable. (I had some issues getting the Julia 1.9-beta to work with my VsCode, which is why I did it this way...)

In summary the PR implements:
- Add @(un)pack functionality, iff UnPack package is loaded (using Requires.jl)
- Add basic unit test for functionality
- Add documentation of feature
- Minor version bump (since new dependence)

### CODE REVIEW

**Comment by lukasgrunwald** [comment]
Bump (the failed nightly runs are not due to any changes made in this commit)

File: README.md [hunk 1]
```diff
@@ -99,6 +99,17 @@ jldopen("example.jld2", "w") do file
 end
 ```
 
+When additionally loading the [UnPack.jl](https://github.com/mauro3/UnPack.jl) package, the there defined `@unpack` and `@pack!` macros can be used to quickly save and load date from  the file-like interface. Example:
```
Comment (JonasIsensee) [inline README.md:102]:
```suggestion
When additionally loading the [UnPack.jl](https://github.com/mauro3/UnPack.jl) package, its `@unpack` and `@pack!` macros can be used to quickly save and load data from the file-like interface. Example:
```

File: README.md [hunk 2]
```diff
@@ -142,6 +153,8 @@ or using slashes as path delimiters:
 @assert load("example.jld2", "mygroup/mystuff") == 42
 ```
 
+The group `file["mygroup"]` can be accessed with the same file-like interface as the "full" struct. When using [UnPack.jl](https://github.com/mauro3/UnPack.jl), the macros `@unpack` and `@pack!` work without modifications.
+
```
Comment (JonasIsensee) [inline README.md:157]: I think I'd prefer to have a separate section in the docs / readme titled something like `UnPack.jl API` 
rather than having it interleaved with the rest.

Personally, I haven't used these macros even once since the regular destructuring syntax was introduced.
(which can't really use here since it uses `getproperty`)

**Review by JonasIsensee — COMMENTED** [review]
Hi @lukasgrunwald ,

thank you for contribution!

File: README.md (hunk 2 above)
Comment (lukasgrunwald) [reply README.md]: Alright, good idea. Your suggestion is implemented in the new commit :+1:

File: README.md (hunk 1 above)
Comment (lukasgrunwald) [reply README.md]: implemented in the new commit

**Review by JonasIsensee — APPROVED** [review]
Looks good to me! :)
